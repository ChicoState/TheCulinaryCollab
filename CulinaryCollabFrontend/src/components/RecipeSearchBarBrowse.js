import React, { useState } from 'react';

const RecipeSearchBarBrowse = ({ recipes, onView }) => {
	const [searchTerm, setSearchTerm] = useState('');
	const [searchResults, setSearchResults] = useState([]);

	const handleInputChange = (e) => {
		const term = e.target.value;
		setSearchTerm(term);
		if (term.length > 0 || term.length === 0) {
			handleSearch(term);
		}
	};
	const handleSearch = (e) => {
		if (e.length <= 1) {
			setSearchResults([]);
			return;
		}
		else {
			const lowercasedTerm = searchTerm.toLowerCase();
			const filtered = recipes.filter(
				(recipe) =>
				(recipe.name && recipe.name.toLowerCase().includes(lowercasedTerm)) ||
				(recipe.ingredients &&
					recipe.ingredients.some((ingredient) => ingredient.ingredient.toLowerCase().includes(lowercasedTerm)))
			);
			console.log(searchTerm);
			setSearchResults(filtered);
		}
	};
	return (
		<div>
		<div className="search-bar-container">
		<input
		type="text"
		placeholder="Search for recipe"
		value={searchTerm}
		onChange={handleInputChange}
		className="search-bar"
		/>
		<button onClick={handleSearch} className="search-button">Search</button>
		</div>
		<ul className="search-results-list">
		{searchResults.map((recipe, index) => (
			<li key={index} className="search-result-item">
			{recipe.name}
			<button onClick={() => onView(recipe)} className="view-recipe-button">View</button>
			</li>
		))}
		</ul>
		</div>
	);
};
export default RecipeSearchBarBrowse;
