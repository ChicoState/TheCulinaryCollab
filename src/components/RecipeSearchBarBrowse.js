import React, { useState } from 'react';

const RecipeSearchBarBrowse = ({ recipes, onView }) => {
	const [searchTerm, setSearchTerm] = useState('');
	const [searchResults, setSearchResults] = useState([]);

	const handleInputChange = (e) => {
		const term = e.target.value;
		setSearchTerm(term);
		if (term.length > 0 || term.length === 0) {
			handleSearch(term);
		}
	};

	const handleSearch = (e) => {
		if (e.length <= 1) {
			setSearchResults([]);
			return;
		} else {
			const lowercasedTerm = searchTerm.toLowerCase();
			const filtered = recipes.filter((recipe) => {
				const recipeNameMatches = recipe.name && recipe.name.toLowerCase().includes(lowercasedTerm);
				const ingredientMatches = recipe.ingredients && recipe.ingredients.some((ingredient) => ingredient.ingredient.toLowerCase().includes(lowercasedTerm));
				const createdByMatches = recipe.createdBy && recipe.createdBy.username && recipe.createdBy.username.toLowerCase().includes(lowercasedTerm);
				return recipeNameMatches || ingredientMatches || createdByMatches;
			});
			setSearchResults(filtered);
		}
	};
	const getRecipeCreator = (recipe) => {
		return recipe.createdBy && recipe.createdBy.username
			? `Created by ${recipe.createdBy.username}`
			: "Public Recipe";
	};

	return (
		<div>
		<div className="search-bar-container">
		<input
		type="text"
		placeholder="Search for recipe"
		value={searchTerm}
		onChange={handleInputChange}
		className="search-bar"
		/>
		</div>
		<ul className="search-results-list">
		{searchResults.map((recipe, index) => (
			<li key={index} className="search-result-item">
			{recipe.name}
			<button onClick={() => onView(recipe)} className="view-recipe-button">View</button>
			<span className="recipe-creator"> {getRecipeCreator(recipe)}</span> 
			</li>
		))}
		</ul>
		</div>
	);
};

export default RecipeSearchBarBrowse;

