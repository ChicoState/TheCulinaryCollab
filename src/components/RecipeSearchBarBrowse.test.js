import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import RecipeSearchBarBrowse from './RecipeSearchBarBrowse';

// Mock recipe data for testing
const recipes = [
  {
    name: 'Rusty Nail',
    ingredients: [
      {
        amount: 4.5,
        ingredient: 'Scotch Whisky',
        unit: 'cl',
      },
      {
        amount: 2.5,
        ingredient: 'Drambuie',
        unit: 'cl',
      },
    ],
    preparation: 'Stirred',
    taste: 'Boozy',
    timing: 'After dinner',
    createdBy: { username: 'chef123' },
  },
  {
    name: 'Mojito',
    ingredients: [
      {
        amount: 4.5,
        ingredient: 'Scotch Whisky',
        unit: 'cl',
      },
      {
        amount: 2.5,
        ingredient: 'Drambuie',
        unit: 'cl',
      },
    ],
    preparation: 'Stirred',
    taste: 'Boozy',
    timing: 'After dinner',
    createdBy: { username: 'chef12' },
  },
];

describe('RecipeSearchBarBrowse', () => {
  test('renders the component', () => {
    const { container } = render(<RecipeSearchBarBrowse recipes={recipes} onView={() => {}} />);
    expect(container).toMatchSnapshot();
  });

  test('handles input change and triggers search', () => {
    const { getByPlaceholderText } = render(<RecipeSearchBarBrowse recipes={recipes} onView={() => {}} />);

    const input = getByPlaceholderText('Search for recipe');
    fireEvent.change(input, { target: { value: 'Rusty Nail' } });

    expect(input.value).toBe('Rusty Nail');

  });

  test('handles search button click', () => {
    const onViewMock = jest.fn();
    const { getByPlaceholderText, getByText } = render(
      <RecipeSearchBarBrowse recipes={recipes} onView={onViewMock} />
    );

    const input = getByPlaceholderText('Search for recipe');
    fireEvent.change(input, { target: { value: 'Rusty Nail' } });

    const viewButton = queryAllByText('View');
    fireEvent.click(viewButton);

    expect(onViewMock).toHaveBeenCalledWith(recipes[0]);
    
  });
});

