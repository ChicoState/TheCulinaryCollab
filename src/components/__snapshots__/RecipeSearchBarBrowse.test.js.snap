// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeSearchBarBrowse renders the component 1`] = `
<div>
  <div>
    <div
      class="search-bar-container"
    >
      <input
        class="search-bar"
        placeholder="Search for recipe"
        type="text"
        value=""
      />
    </div>
    <ul
      class="search-results-list"
    />
  </div>
</div>
`;
